Add WN hardware recommendations to the WN overview (SOFTWARE-3577) (#503)
opensciencegrid/docs: docs/worker-node/using-wn.md

If you would like to test the minimal OS environment that jobs can expect, you can test out your scientific software in [the OSG Docker image](https://hub.docker.com/r/opensciencegrid/osg-wn/).

> Suggested Hardware Requirements on Worker Nodes

Reviewer: ```suggestion
Hardware Recommendations
```

------------------------
| Hardware               | Minimum | Recommended                         | Notes                                             |
|:-----------------------|:--------|:----------------------|:--------------------------------------------------|

> |Core per pilot                   |  1      |8                      | Dependends on the supported VOs. The total core count on every node in the cluster must be divisible by core per pilot.|

Reviewer: Spelling
```suggestion
|Core per pilot                   |  1      |8                      | Depends on the supported VOs. The total core count on every node in the cluster must be divisible by core per pilot.|
```

> |Core per pilot                   |  1      |8                      | Dependends on the supported users. The total core count on every node in the cluster must be divisible by core per pilot.|

Reviewer: Huh, is the second sentence true? I didn't know that.

```suggestion
|Core per pilot                   |  1      |8                      | Depends on the supported VOs. The total core count on every node in the cluster must be divisible by core per pilot.|
```

Author: @brianhlin of course otherwise you end up in suggestions you don't want to be. Either cores never utilized or a continuously fragmented pool

> |Memory per core                 | 1024MB  | 2048MB                  | Memory per core times core per pilot should be less than the total memory on every node. Do not overcommit |

Reviewer: ```suggestion
|Memory per core                 | 1024MB  | 2048MB                  | Memory per core times core per pilot needs to be less than the total memory on every node. Do not overcommit. |
```

> |Scratch disk per core ([OSG_WN_TMP](/worker-node/using-wn/#osg_wn_tmp))| 2GB    | 10 GB                  | This can be overcommitted if a mix of different VO jobs is expected.|

Reviewer: Minor style suggestion

```suggestion
|Scratch disk per core ([OSG_WN_TMP](#osg_wn_tmp))| 2GB    | 10 GB                  | This can be overcommitted if a mix of different VO jobs is expected.|
```

> |Scratch disk per core([OSG_WN_TMP](/worker-node/using-wn/#osg_wn_tmp))| 2GB    | 10 GB                  | This can be overcommitted if a mix of different VO jobs is expected|

Reviewer: ```suggestion
|Scratch disk per core ([OSG_WN_TMP](/worker-node/using-wn/#osg_wn_tmp))| 2GB    | 10 GB                  | This can be overcommitted if a mix of different VO jobs are expected|
```

> |CVMFS [Cache](/worker-node/install-cvmfs/#before-starting) | 10 GB | 20 GB | This is a value per node and not per core |

Reviewer: ```suggestion
|CVMFS [Cache](/worker-node/install-cvmfs/#before-starting) (optional)| 10 GB | 20 GB | This is a value per node and not per core |
```
